File: online-judge-solutions/Codeforces/1085A.cpp
//https://codeforces.com/problemset/problem/1085/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	0 KB
#include<iostream>
using namespace std;
int main(){
	string s;
	while(cin>>s){
		string w;
		int t=0;
		while(s.size()>0){
			int n=s.size();
			int i=(t%2==0?(n-1):n)/2;
			w+=s[i];
			s.erase(i,1);
			t++;
		}
		cout<<w<<endl;
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1085A.py
#https://codeforces.com/problemset/problem/1085/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
s,t,w=input(),0,''
while(len(s)>0):
    n=len(s)
    i=int((n-1 if t%2==0 else n)/2)
    w+=s[i]
    s=s[0:i]+s[i+1:]
    t+=1
print(w)
